draft-4/CommandLineTool.yml

```diff
@@ -626,6 +626,44 @@ $graph:
         Docker container.
 
 
+- type: record
+  name: SoftwareRequirement
+  extends: "#ProcessRequirement"
+  doc: |
+    Software that should be configured in the environment of the defined
+    process.
+
+    While absolute portability between platforms can be achieved with container
+    technologies like Docker, this mechanism provides an inexact and imprecise
+    fallback. The inexact nature is due to this standard not specifying the
+    base operating environment of the process or how the that environment is to
+    be configured as a result of these hints. The imprecise nature of these
+    hints is due to the `name` and `version` fields being unstructured and not
+    namespaced. The optional `identifier` field is recommended so that users
+    and platforms can distinguish between software that has the same name.
+  fields:
+    - name: spec
+      type:
+        type: array
+        items: SoftwareSpec
+      doc: "The list of software to be configured."
+
+- name: SoftwareSpec
+  type: record
+  fields:
+    - name: name
+      type: string
+      doc: "The name of the software to be configured."
+      jsonldPredicate: "@id"
```

Comment:
I don't know if this is the correct thing here; was the only was to process the spec due to predicate collision. The other option I thought of was to link it to `doap:name` but that didn't resolve the predicate error.